### PR TITLE
fix: fw-1618 my profile url

### PR DESCRIPTION
**Pull request by swkatmask**
https://mask.atlassian.net/browse/fw-1618

### CODE REVIEW

File: src/app/(normal)/pages/Profile.tsx [hunk 1]
```diff
@@ -33,9 +33,10 @@ interface ProfilePageProps {
 export function ProfilePage({ profiles }: ProfilePageProps) {
     const currentTwitterProfile = useTwitterStateStore.use.currentProfile();
     const { source, identity } = ProfileContext.useContainer();
+    const myProfiles = useMyAllProfiles();
 
     const pathname = usePathname();
-    const isOtherProfile = pathname !== PageRoute.Profile && isRoutePathname(pathname, PageRoute.Profile);
+    const isOtherProfile = !myProfiles.some((x) => x.source === source && x.identity === identity);
```
Comment (guanbinrui) [inline src/app/(normal)/pages/Profile.tsx:39]: isSameProfile

File: src/app/(normal)/profile/page.tsx [hunk 2]
```diff
@@ -19,33 +18,8 @@ export default function Page() {
...
-            return FireflySocialMediaProvider.getAllPlatformProfiles(
-                currentLensProfile?.handle,
-                currentFarcasterProfile?.profileId,
-                currentTwitterProfile?.profileId,
-            );
-        },
-    });
-
     const defaultProfiles = useMemo(() => {
         return compact([
```
Comment (guanbinrui) [inline src/app/(normal)/profile/page.tsx:22]: Sort against `SORTED_SOCIAL_SOURCES`.

File: src/app/(normal)/profile/pages/DetailPage.tsx [hunk 3]
```diff
@@ -19,15 +20,22 @@ interface Props {
 export function ProfileDetailPage({ identity, source }: Props) {
     const currentSource = resolveSourceFromUrl(source);
 
-    const { data: profiles = EMPTY_LIST, isLoading } = useQuery({
+    const myProfiles = useMyAllProfiles();
+    const isMyProfile = myProfiles.some((profile) => profile.source === currentSource && profile.identity === identity);
```
Comment (guanbinrui) [inline src/app/(normal)/profile/pages/DetailPage.tsx:24]: isSameProfile

File: src/components/SideBar/Menu.tsx [hunk 4]
```diff
@@ -54,176 +57,177 @@ export const Menu = memo(function Menu({ collapsed = false }: MenuProps) {
...
+
+    const myProfiles = useMyAllProfiles();
 
-    const checkIsSelected = (href: `/${string}`) =>
-        href === PageRoute.Profile ? isMyProfile || pathname === PageRoute.Profile : isRoutePathname(pathname, href);
+    const checkIsSelected = (href: `/${string}`) => {
+        if (isRoutePathname(href, '/profile')) {
+            const identity = isRoutePathname(pathname, '/profile') ? pathname.split('/')[2] ?? '' : '';
+            const isMyProfile = myProfiles.some(
+                (profile) => profile.source === currentSource && profile.identity === identity,
```
Comment (guanbinrui) [inline src/components/SideBar/Menu.tsx:70]: isSameProfile

File: src/hooks/useMyAllProfiles.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    const { data: profiles } = useQuery({
+        enabled,
+        queryKey: ['all-profiles', 'myself', lensHandle, farcasterProfileId, twitterId],
+        queryFn: async () => {
+            return FireflySocialMediaProvider.getAllPlatformProfiles(lensHandle, farcasterProfileId, twitterId);
+        },
+    });
+
+    const defaultProfiles = useMemo(() => {
+        return compact([
```
Comment (guanbinrui) [inline src/hooks/useMyAllProfiles.ts:27]: Sort against `SORTED_SOCIAL_SOURCES`.

File: src/hooks/useMyAllProfiles.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import { useMemo } from 'react';
+
+import { Source } from '@/constants/enum.js';
+import { FireflySocialMediaProvider } from '@/providers/firefly/SocialMedia.js';
+import { useFarcasterStateStore, useLensStateStore, useTwitterStateStore } from '@/store/useProfileStore.js';
+
+export function useMyAllProfiles(enabled = true) {
+    const lensProfile = useLensStateStore.use.currentProfile();
+    const farcasterProfile = useFarcasterStateStore.use.currentProfile();
+    const twitterProfile = useTwitterStateStore.use.currentProfile();
```
Comment (guanbinrui) [inline src/hooks/useMyAllProfiles.ts:12]: useCurrentProfile

File: src/hooks/useActiveProfiles.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import { useFarcasterStateStore, useLensStateStore, useTwitterStateStore } from '@/store/useProfileStore.js';
+
+export function useActiveProfiles() {
+    const currentLensProfile = useLensStateStore.use.currentProfile();
+    const currentFarcasterProfile = useFarcasterStateStore.use.currentProfile();
+    const currentTwitterProfile = useTwitterStateStore.use.currentProfile();
+    return useMemo(() => {
+        return compact([currentFarcasterProfile, currentLensProfile, currentTwitterProfile]);
+    }, [currentLensProfile, currentFarcasterProfile, currentTwitterProfile]);
+}
```
Comment (guanbinrui) [inline src/hooks/useActiveProfiles.ts:13]:
```suggestion
export function useActiveProfileAll() {
    const curentProfileAll = useCurrentProfileAll()
    return useMemo(() => {
        return compact(SORTED_SOCIAL_SOURCE.map(x => currentProfileAll[x]))
    }, [currentProfileAll]);
}
```

File: src/app/(normal)/profile/pages/DetailPage.tsx (hunk 3 above)
Comment (swkatmask) [reply src/app/(normal)/profile/pages/DetailPage.tsx:24]: ditto.

File: src/app/(normal)/pages/Profile.tsx (hunk 1 above)
Comment (swkatmask) [reply src/app/(normal)/pages/Profile.tsx:39]: `myProfile` is `FireFlyProfile`, different from `Profile`

image

File: src/components/SideBar/Menu.tsx (hunk 4 above)
Comment (swkatmask) [reply src/components/SideBar/Menu.tsx:70]: ditto

**Pull request merged**